izumi-reflect/izumi-reflect/src/test/scala-3/izumi/reflect/test/TagTest.scala

```diff
@@ -23,6 +25,11 @@ class TagTest extends SharedTagTest with TagAssertions {
       assertNotChild(Tag[Animal | String].tag, Tag[Dog | String].tag)
     }
 
+    "Does not synthesize Tags for abstract types, but recursively summons" in {
```

Comment:
This should be part of `SharedTagTest` I believe, to ensure the behavior is the same in Scala 2 & 3. You may also want to test against types `object X; X.T`, `val x; x.T` & `trait X; X#T` since they're all quite different types than `this.T` aand Tag does not behave uniformly wrt them all